src/sphinx_substitution_extensions/utils.py

```diff
@@ -0,0 +1,8 @@
+def exists_dependency(
```

Comment:
This change means that someone can do:

```python
from sphinx_substitution_extensions.utils import exists_dependency
```

Which adds to what looks like a public interface.
Therefore, I would prefer to use `_utils.py`.

However:

* I am not fond of the name `utils` at all. I would prefer to have a file name which refers only to this function than to create `utils` (or `common` or `general` or anything similar).
* I don't think that it is necessary to have this function here - it can be a private function in `__init__.py`.